src/matrixUtils.js

```diff
@@ -54,11 +54,6 @@ function parseHostnameAndPort(serverName) {
 async function discoverHomeserverUrl(serverName) {
     let {hostname, port, defaultPort} = parseHostnameAndPort(serverName);
 
-    // Don't continue if we consider the hostname part to resolve to our blacklisted IP ranges
```

Comment:
This check must not be removed, otherwise we would allow IP's or domains with an explicit port even if they are in our blacklist.

I realize we also do this later in the `axiosGet` as well, but instead of removing this check it would be nicer to skip it based on the environment variable, so the functionality of this `discoverHomeserverUrl` function stays the same.

Reply:
> This check must not be removed, otherwise we would allow IP's or domains with an explicit port even if they are in our blacklist.
I added the check again.

> I realize we also do this later in the `axiosGet` as well, but instead of removing this check it would be nicer to skip it based on the environment variable, so the functionality of this `discoverHomeserverUrl` function stays the same.
I would assume, skipping the check here is another PR as it would affect the token authentication and not the room membership and power level checks? The semantic of the UVS_DISABLE_IP_BLACKLIST currently does not cover the token verification. Thought about this for a while: I assume, this skip would make sens in a deployment without federation where we could have no users from third-party homeservers in a room and the UVS running only for that HS. However, in this situation, we would also need to implement to check only against the host defined by UVS_OPENID_VERIFY_SERVER_NAME (which could then be a private IP).